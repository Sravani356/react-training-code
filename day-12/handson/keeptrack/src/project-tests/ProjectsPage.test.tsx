import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';

import Projectspage from '../project/ProjectsPage';

import { QueryClientProvider, QueryClient } from 'react-query';
const queryClient = new QueryClient();


describe('<ProjectsPage/>', () => {

test('rendering ProjectsPage component', () => {
     render(
  <QueryClientProvider client={queryClient}>
    <Projectspage/>
    </QueryClientProvider>
    );
    const heading = screen.getByTestId("project-page");
    expect(heading).toHaveTextContent("Projects");
});

test('snapshot', () => {
    const tree = renderer.create(
  <QueryClientProvider client={queryClient}>
  <Projectspage/>
  </QueryClientProvider>).toJSON()
expect(tree).toMatchSnapshot();
});
});

// test('rendering ProjectsPage component', () => {
//     render(<ProjectCard project={Project} onEdit={jest.fn()/>)
// });

// test('rendering ProjectsPage component', () => {

//     interface ProjectListProps { //to define the props that come into the component
//         projects: Project[];
//         onSave: (project: Project) => void;
//       }
   
//     const saveProject = jest.fn();

//     render(<ProjectList />); 
    
// })