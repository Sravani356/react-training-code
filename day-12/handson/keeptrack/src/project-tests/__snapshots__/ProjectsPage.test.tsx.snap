// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsPage/> snapshot 1`] = `
<div
  data-testid="project-page"
>
  <h1>
    Projects
  </h1>
  <div
    className="center-page"
  >
    <span
      className="spinner primary"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;
